library/init/meta/simp_tactic.lean

```diff
@@ -455,12 +457,10 @@ meta def join_user_simp_lemmas_core : simp_lemmas → list name → tactic simp_
 | S []             := return S
 | S (attr_name::R) := do S' ← get_user_simp_lemmas attr_name, join_user_simp_lemmas_core (S.join S') R
 
-meta def join_user_simp_lemmas (no_dflt : bool) (attrs : list name) : tactic simp_lemmas :=
-if no_dflt then
-  join_user_simp_lemmas_core simp_lemmas.mk attrs
-else do
-  s ← simp_lemmas.mk_default,
-  join_user_simp_lemmas_core s attrs
+meta def join_user_simp_lemmas (no_dflt : bool) (attrs : list name) : tactic simp_lemmas := do
+s ← simp_lemmas.mk_default,
+let s := if no_dflt then s.erase_simp_lemmas else s,
+join_user_simp_lemmas_core s attrs
```

Comment:
While we don't really have a consistent style for `do` blocks, this file seems to use
```suggestion
meta def join_user_simp_lemmas (no_dflt : bool) (attrs : list name) : tactic simp_lemmas :=
do s ← simp_lemmas.mk_default,
   let s := if no_dflt then s.erase_simp_lemmas else s,
   join_user_simp_lemmas_core s attrs
```
or
```suggestion
meta def join_user_simp_lemmas (no_dflt : bool) (attrs : list name) : tactic simp_lemmas :=
do
  s ← simp_lemmas.mk_default,
  let s := if no_dflt then s.erase_simp_lemmas else s,
  join_user_simp_lemmas_core s attrs
```
either of which look better to me since the `do` is more obvious.